ls1intum/Artemis
# Feature/guided tour/automatic assignment of tutorial groups

### Checklist

### Motivation and Context


Using the ArTEMiS internal user management, each new user have to be manually assigned to the interactive tutorial course in order to enable it for him/her. These changes automatically add (if included in the configuration files) these tutorial course groups to a new user during its creation. The groups will be assigned based on the user's given roles (USER, TA, INSTRUCTOR). 

### Description


- Added new repository queries to retrieve a course by its students- tutors- instructors group name.
- Added new properties  (`info.tutorial-course-groups.<students, tutors, instructors>`) where the group names of the tutorial course should be placed
- Added a `addTutorialGroups(ManagedUserVM user)` method which is executed if `artemis.user-management.use-external` is set to false and adds the relative tutorial course groups to the new user if these are present in the before mentioned properties.


### Steps for Testing


1. Create a course in ArTEMiS
2. Insert the course short name in `info.guided-tour.courseShortName` to enable the tutorial
3. Insert the course group names in `info.tutorial-course-groups.<students, tutors, instructors>`
4. Log in to Artemis
5. Navigate to User Management
6. Create a new user
7. Check if the new created user has been added to the tutorial groups (Example: a new user with _USER_ and _INSTRUCTOR_ roles should be added to the _students_ and _instructors_ groups)

### Test Coverage

* UserService.java: 83%

### Screenshots
